checks/permissions.go

```diff
@@ -24,149 +24,251 @@ import (
 	sce "github.com/ossf/scorecard/v2/errors"
 )
 
-const CheckPermissions = "Token-Permissions"
+const CheckTokenPermissions = "Token-Permissions"
 
 //nolint:gochecknoinits
 func init() {
-	registerCheck(CheckPermissions, leastPrivilegedTokens)
+	registerCheck(CheckTokenPermissions, TokenPermissions)
 }
 
-func leastPrivilegedTokens(c *checker.CheckRequest) checker.CheckResult {
-	r, err := CheckFilesContent(".github/workflows/*", false, c, validateGitHubActionTokenPermissions)
-	return createResultForLeastPrivilegeTokens(r, err)
+// Holds stateful data to pass thru callbacks.
+// Each field correpsonds to a GitHub permission type, and
+// will hold true if declared non-write, false otherwise.
+type permissionCbData struct {
+	permissions map[string]bool
```

Comment:
nit: s/permissions/writePermissions/ or something similar makes this easier to understand.

Reply:
done.